### PR TITLE
Add overload to AddAzureBlobStorage to supply your own BlobServiceClient from delegate

### PR DESCRIPTION
**Pull request by willadsen**
BlobServiceClient

**What this PR does / why we need it**:
If the application needs to monitor a `BlobServiceClient` which is registered ie. using named instances (see https://learn.microsoft.com/en-us/dotnet/azure/sdk/dependency-injection)
```
builder.Services.AddAzureClients(clientBuilder =>
{
    clientBuilder.AddBlobServiceClient(builder.Configuration.GetSection("PublicStorage"));
    clientBuilder.AddBlobServiceClient(builder.Configuration.GetSection("PrivateStorage"))
        .WithName("PrivateStorage");
});
```
then this PR tries to solve that issue by overloading the `AddAzureBlobStorage` extension method with a `Func<IServiceProvider,BlobServiceClient>` function argument to be able to supply your own way of resolving the `BlobServiceClient` instance.

**Which issue(s) this PR fixes**:
No issue created, should I make one?

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
Only if the user should wish to use these new methods

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [X] Code compiles correctly
- [X] Created/updated tests
- [X] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by willadsen** [comment]
@dotnet-policy-service agree company="TrackMan"

**Comment by willadsen** [comment]
@dotnet-policy-service agree company="TrackMan"

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs:165]:
```suggestion
    /// A <see cref="BlobServiceClient"/> service must be registered in the service container. For named instances
    /// you may use other overload with <see cref="Func<IServiceProvider, BlobServiceClient>"/> argument.

```

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs:167]:
```suggestion
```

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs:204]:
```suggestion
```

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs:203]:
```suggestion
    /// <param name="clientFactory">Delegate for creating a <see cref="BlobServiceClient"/>.</param>
```

File: test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs
Comment (sungam3r) [inline test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs:178]:
```suggestion
```

File: test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs
Comment (sungam3r) [inline test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs:93]:
```suggestion

```

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs:148]:
```suggestion
        return builder.Add(new HealthCheckRegistration(
```

File: test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs
Comment (sungam3r) [inline test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs:148]:
```suggestion

```

File: test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs
Comment (sungam3r) [inline test/HealthChecks.AzureStorage.Tests/DependencyInjection/AzureBlobStorageRegistrationTests.cs:93]:
```suggestion

    [Theory]
```

**Review by sungam3r — APPROVED** [review]

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs:166]:
```suggestion
    /// you may use other overload with <see cref="Func{IServiceProvider, BlobServiceClient}"/> argument.
```

**Comment by sungam3r** [comment]
@willadsen Thanks. I'm fine to merge after updating API approval file.

**Comment by cieciurm** [comment]
Hi @willadsen,

Are you planning to finish up this PR this week?
I see there is very small changes required. Perhaps you could make it? 😄

**Comment by willadsen** [comment]
> Hi @willadsen,
> 
> 
> 
> Are you planning to finish up this PR this week?
> 
> I see there is very small changes required. Perhaps you could make it? 😄 

Unfortunately no, as I am on vacation. It'll be a couple of weeks until I will be able to complete the PR. Feel free to complete it 😎

**Comment by sungam3r** [comment]
Superseded by #1889.

**Pull request closed without merging**
